### PR TITLE
Update esp32 platform to 2024.04.14

### PR DESCRIPTION
**Pull request by Jason2866**
## Description:

Updated HWCDC code in Arduino. Should fix boot stuck issues with ESP32-S3

@fyi @s-hadinger 

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [ ] The code change is tested and works with Tasmota core ESP8266 V.2.7.6
  - [x] The code change is tested and works with Tasmota core ESP32 V.3.0.0
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/tasmota.ino [hunk 1]
```diff
@@ -474,10 +474,10 @@ void setup(void) {
 
   bool is_connected_to_USB = false;
 #if SOC_USB_SERIAL_JTAG_SUPPORTED  // Not S2
-  for (uint32_t i = 0; i < 5; i++) {  // wait up to 250 ms - maybe a shorter time is enough
-      is_connected_to_USB = usb_serial_jtag_is_connected();
+  for (uint32_t i = 0; i < 5; i++) {  // wait up to 50 ms - maybe a shorter time is enough
+      is_connected_to_USB = HWCDC::isPlugged();
```
Comment (Staars) [inline tasmota/tasmota.ino:478]: This change resolves to:
  inline static bool isPlugged(void) {
    return usb_serial_jtag_is_connected();
  }

So what is the purpose?

File: tasmota/tasmota.ino [hunk 2]
```diff
@@ -474,10 +474,10 @@ void setup(void) {
...
       if (is_connected_to_USB) { break; }
-      delay(50);
+      delay(10);
```
Comment (Staars) [inline tasmota/tasmota.ino:480]: Are you sure, that the timeout is now magically shorter? I would expect undefined behavior now.

**Comment by Jason2866** [comment]
Does not fix at all. Still issues with S3.

**Comment by Staars** [comment]
The most recent code changes in Arduino's HWCDC class require some refactoring in Tasmotas USB connection check, if we want to use Arduino's API.

New way is:
1. Always start USB CDC console.
2. Check for plugged USB cable using the Arduino API
3. a) if USB connection is detected proceed with already initialised HWCDC console
    b) if NO connection is detected - deinit HWCDC and fall back to serial.

**Pull request merged**